src/BlazorWebView/src/Maui/Android/AndroidMauiAssetFileProvider.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Android.Content.Res;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.WebView.Maui
+{
+	/// <summary>
+	/// A minimal implementation of an IFileProvider to be used by the BlazorWebView and WebViewManager types.
+	/// </summary>
+	internal sealed class AndroidMauiAssetFileProvider : IFileProvider
+	{
+		private readonly AssetManager _assets;
+		private readonly string _contentRootDir;
+
+		public AndroidMauiAssetFileProvider(AssetManager? assets, string contentRootDir)
+		{
+			_assets = assets ?? throw new ArgumentNullException(nameof(assets));
+			_contentRootDir = contentRootDir;
+		}
+
+		public IDirectoryContents GetDirectoryContents(string subpath)
+			=> new AndroidMauiAssetDirectoryContents(_assets, Path.Combine(_contentRootDir, subpath));
+
+		public IFileInfo GetFileInfo(string subpath)
+			=> new AndroidMauiAssetFileInfo(_assets, Path.Combine(_contentRootDir, subpath));
+
+		public IChangeToken? Watch(string filter)
+			=> null;
+
+		private sealed class AndroidMauiAssetFileInfo : IFileInfo
+		{
+			private readonly AssetManager _assets;
+			private readonly string _filePath;
+			private readonly Lazy<bool> _lazyAssetExists;
+			private readonly Lazy<long> _lazyAssetLength;
+
+
+			public AndroidMauiAssetFileInfo(AssetManager assets, string filePath)
+			{
+				_assets = assets;
+				_filePath = filePath;
+
+				Name = Path.GetFileName(filePath);
+
+				_lazyAssetExists = new Lazy<bool>(() =>
+				{
+					try
+					{
+						using var stream = _assets.Open(_filePath);
+						return true;
+					}
+					catch
+					{
+						return false;
+					}
+				});
+
+
+				_lazyAssetLength = new Lazy<long>(() =>
+				{
+					try
+					{
+						// The stream returned by AssetManager.Open() is not seekable, so we have to read
+						// the contents to get its length. In practice, Length is never called by BlazorWebView,
+						// so it's here "just in case."
+						using var stream = _assets.Open(_filePath);
+						using var memoryStream = new MemoryStream();
+						stream.CopyTo(memoryStream);
+						return memoryStream.Length;
```

Comment:
Just a note this is quite heavy, especially if this is a large file. Since it probably will never be called, it is not essential. However, this might be worth reading via a buffer of say 4K and then just keep reading into that.

```csharp
var buffer = ArrayPool<byte>.Shared.Rent(4096);
long length = 0;
while (length != (length += source.Read(buffer))) {
	// loop
}
ArrayPool<byte>.Shared.Return(buffer);

Reply:
Will update. I wish on Android there was a legit way to do this. You can't access the file system, calling AssetManager.OpenFd() isn't supported with compressed assets (which by default everything is), so you can get only a stream, and count the bytes as they flow by into nothingness :(